src/formats/farbfeld.zig

```diff
@@ -0,0 +1,109 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const mem = std.mem;
+const Allocator = std.mem.Allocator;
+const assert = std.debug.assert;
+const ImageUnmanaged = @import("../ImageUnmanaged.zig");
+const FormatInterface = @import("../FormatInterface.zig");
+const color = @import("../color.zig");
+const utils = @import("../utils.zig");
+const buffered_stream_source = @import("../buffered_stream_source.zig");
+
+pub const ReadError = ImageUnmanaged.ReadError;
+pub const WriteError = ImageUnmanaged.WriteError || error{BigDimensions};
+
+pub const Header = extern struct {
+    const width_size = 4;
+    const height_size = 4;
+    const size = 16;
+    const magic_value: [8]u8 = [8]u8{ 'f', 'a', 'r', 'b', 'f', 'e', 'l', 'd' };
+
+    width: u32 align(1),
+    height: u32 align(1),
+
+    fn encode(header: Header) [size]u8 {
+        var result: [size]u8 = undefined;
+        @memcpy(result[0..8], &magic_value);
+        std.mem.writeInt(u32, result[8..12], header.width, .big);
+        std.mem.writeInt(u32, result[12..16], header.height, .big);
+        return result;
+    }
+    comptime {
+        assert(size == 16);
+    }
+};
+
+pub fn formatInterface() FormatInterface {
+    return FormatInterface{
+        .format = format,
+        .formatDetect = formatDetect,
+        .readImage = readImage,
+        .writeImage = writeImage,
+    };
+}
+
+pub fn format() ImageUnmanaged.Format {
+    return ImageUnmanaged.format.farbfeld;
+}
+/// Taken a stream, Returns true if and only if the stream contains the magic value "fabfeld"
+pub fn formatDetect(stream: *ImageUnmanaged.Stream) ReadError!bool {
+    var magic_buffer: [Header.magic_value.len]u8 = undefined;
+    const bytes_read = try stream.read(magic_buffer[0..]);
+    return bytes_read == Header.magic_value.len and
+        mem.eql(u8, magic_buffer[0..], Header.magic_value[0..]);
+}
+/// consume stream and returns ImageUnmanaged as farbfeld format. pixelformat is rgba64 and does not have animation. Caller owns returned memory. https://tools.suckless.org/farbfeld/
+pub fn readImage(allocator: std.mem.Allocator, stream: *ImageUnmanaged.Stream) ReadError!ImageUnmanaged {
```

Comment:
`readImage` and `writeImage` should only be adapter functions for the Format interface. There is no way to read or write the pixel storage if you want to bypass `Image` and use the `farbfeld` format directly. See the design philosophy in the README file.